pkg/controller/controller_instance.go

```diff
@@ -203,10 +210,34 @@ func (c *controller) reconcileInstanceDelete(instance *v1alpha1.Instance) error
 	return nil
 }
 
+// isInstanceFailed returns whether the instance has a failed condition with
+// status true.
+func isInstanceFailed(instance *v1alpha1.Instance) bool {
+	for _, condition := range instance.Status.Conditions {
+		if condition.Type == v1alpha1.InstanceConditionFailed && condition.Status == v1alpha1.ConditionTrue {
+			return true
+		}
+	}
+
+	return false
+}
+
 // reconcileInstance is the control-loop for reconciling Instances. An
 // error is returned to indicate that the binding has not been fully
 // processed and should be resubmitted at a later time.
 func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
+	// Currently, we only set a failure condition if the initial provision
+	// call fails, so if that condition is set, there is no more work to do.
+	// We will need to reevaluate this logic as we make any changes to capture
```

Comment:
@nilebox I don't want to try to fully solve orphan mitigation in this PR because it touches every facet of provision,deprovision,update,last_operation, etc.  This fixes the immediate need (and bug that hurts the worst) of making the catalog no longer retry failed provisions.  Orphan mitigation is significantly more involved, so I want to lay the groundwork here and do orphan mitigation in some follow-ups.

I expect orphan mitigation will probably be a significantly larger PR than this one and might itself have to be split into more pieces.